pkg/rpc/request/param.go

```diff
@@ -210,20 +211,46 @@ func (p *Param) GetIntStrict() (int, error) {
 	if p.IsNull() {
 		return 0, errNotAnInt
 	}
-	if p.cache == nil {
-		var i int
-		err := json.Unmarshal(p.RawMessage, &i)
-		if err != nil {
-			return i, errNotAnInt
-		}
-		p.cache = i
+	value, err := p.fillIntCache()
+	if err != nil {
+		return 0, err
 	}
-	if i, ok := p.cache.(int); ok {
-		return i, nil
+	if i, ok := value.(int64); ok && i == int64(int(i)) {
+		return int(i), nil
 	}
 	return 0, errNotAnInt
 }
 
+func (p *Param) fillIntCache() (interface{}, error) {
+	if p.cache != nil {
+		return p.cache, nil
```

Comment:
Then you need to adjust all other cache-related unmarshallers ( `GetString`, `GetBoolean`, etc.) to store `int64` instead of `int` inside the cache after unmarshalling.

Reply:
Fixed.